extensions/podman/packages/extension/src/extension.ts

```diff
@@ -1244,7 +1245,15 @@ export function registerOnboardingRemoveUnsupportedMachinesCommand(): extensionA
   });
 }
 
-export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+async function exec(
+  args: string[],
+  connection?: extensionApi.ProviderContainerConnection,
+  options?: extensionApi.RunOptions,
+): Promise<extensionApi.RunResult> {
+  return execPodman(args, connection?.connection.vmTypeDisplayName, options);
```

Comment:
AFAIK it's not the display name to use but the vmType

Reply:
If I have two wsl vmType, I need to use the connection name to be able to determine which one is supposed to be targeted

Reply:
Both can be used, because if known display name is passed, it would be converted to vmType, otherwise if the passed value cannot be converted, it is used as as vmType (see [execPodman](https://github.com/containers/podman-desktop/blob/87c5065fefc26a7ff207e1f180b01a8fe36be32a/extensions/podman/packages/extension/src/util.ts#L148)).

Reply:
The `CONTAINERS_MACHINE_PROVIDER` allow us to target a provider (`qemu`, `wsl` etc.) but not a specific machine in the provider.

But as Florent said, let's iterate after this PR